modules/message-lane/src/mock.rs

```diff
@@ -176,13 +212,20 @@ impl TestMessageDeliveryAndDispatchPayment {
 		frame_support::storage::unhashed::put(b":reject-message-fee:", &true);
 	}
 
-	/// Returns true if given fee has been paid by given relayer.
+	/// Returns true if given fee has been paid by given submitter.
 	pub fn is_fee_paid(submitter: AccountId, fee: TestMessageFee) -> bool {
 		frame_support::storage::unhashed::get(b":message-fee:") == Some((submitter, fee))
 	}
+
+	/// Returns true if given relayer has been rewarded with given balance. The reward-paid flag is
+	/// cleared after the call.
+	pub fn is_reward_paid(relayer: AccountId, fee: TestMessageFee) -> bool {
+		let key = (b":relayer-reward:", relayer, fee).encode();
+		frame_support::storage::unhashed::take::<bool>(&key).is_some()
```

Comment:
Is using functionality from the `unhashed` module only useful for testing, or does it have applications beyond that?

Reply:
It is also used for some keys that are shared by all runtimes (aka 'well_known_keys') - i.e. ":code", ":max_heap_pages", ...